File: Majority_element/code.cpp
class Solution{
  public:
    int majorityElement(int a[], int size)
    {
        
        int res = -1;
        sort(a, a+size);
        int i = 0;
        int x = 1;
        if(size == 1){
            return a[0];
        }
        while(i<size){
            if(a[i] ==a[i+1]){
                x=1;
                while(a[i] ==a[i+x]){
                    x++;
                }
                if(x>(size/2)){
                    return a[i];
                }
                i+=x;
            }
            else{
                i++;
            }
        }
        return res;
    }
};

File: Majority_element/code.java
class Solution
{
     static int majorityElement(int a[], int n)
    {
        // your code here
        int res=0,count=1;
        
        for(int i=1;i<n;i++)
        {
            if(a[res]==a[i])
            {
                count++;
            }
            else{
                count--;
            }
            
            if(count==0)
            {
                res=i;
                count=1;
            }
            
            
        }
        
        count=0;
        
        for(int i=0;i<n;i++)
        {
            if(a[res]==a[i])
            {
                count++;
            }
        }
            
            if(count<=n/2)
            {
                return -1;
            }
        
        return a[res];
  }
}
